framework/db/ActiveRecord.php

```diff
@@ -281,7 +287,13 @@ public static function find()
      */
     public static function tableName()
     {
-        return '{{%' . Inflector::camel2id(StringHelper::basename(get_called_class()), '_') . '}}';
+        static $tableName = null;
+
+        if ($tableName === null) {
+            $tableName = '{{%' . Inflector::camel2id(StringHelper::basename(get_called_class()), '_') . '}}';
+        }
+
+        return $tableName;
```

Comment:
Does not really make sense.
Method `tableName()` is meant to be overriden by the child class, which should declare table name explicitely.
Still can be kept.